src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -309,17 +311,17 @@ public sealed class ProjectInfo
                 newHostObjectType);
         }
 
-        public ProjectInfo WithDocuments(IEnumerable<DocumentInfo> documents)
+        public ProjectInfo WithDocuments(IEnumerable<DocumentInfo>? documents)
```

Comment:
❔ Do we really want to say this allows null? Answer applies to this and several following `With` methods that take enumerables.